ratelimit/src/lib.rs

```diff
@@ -1,203 +1,293 @@
-// Copyright 2019-2020 Twitter, Inc.
-// Licensed under the Apache License, Version 2.0
-// http://www.apache.org/licenses/LICENSE-2.0
+//! This crate provides a simple implementation of a ratelimiter that can be
+//! shared between threads.
+//!
+//! ```no_run
```

Comment:
I think the documentation could be helped with explanation for the new design, some user guidelines, and a couple of examples.

I think we should confirm that `refill_amount` and `refill_interval` will be applied literally. Explain the range of values that make sense (e.g. nano-second level interval wouldn't make sense on most OS, probably should start with multiple microseconds). `capacity` and `available` (some thoughts on naming later near the `Builder` structure) maps into the maximum and initial token level of the rate limiter, and should be mentioned and explained as such. 

When it comes to user guide, we can use a few examples to show typical use of a rate limiter. E.g. a rate limiter for a (public) API should probably be initialized with full capacity, and the maximum token level is typically the same as `refill_amount`. On the other hand, to mimic the behavior of a linux `qdisc` controller which is burst friendly, the rate limiter could have a maximum capacity several times higher than the `refill_amount`, and start with the full capacity. A third example could be a "TCP slow-start" rate limiter where the bucket starts empty and refills slowly, but with a decent maximum capacity.